### PR TITLE
mame rom-tools 0.244

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
linux build failure

```
brew install --verbose --build-bottle mame
brew install --verbose --build-bottle rom-tools
```

**Comment by cho-m** [comment]
Upstream issue due to missing header for `ptrdiff_t`.

It was fixed in https://github.com/mamedev/mame/commit/034e0d2c87a16e90783f742f39850dc7d85def1f along with other changes.

**Comment by danielnachun** [comment]
I pushed commits to add the upstream commit linked by @cho-m as a patch which fixed the build when I tested it locally.  Once we requeue this to run with `CI-long-timeout` we should hopefully be ready to go.

File: Formula/mame.rb [hunk 1]
```diff
@@ -55,6 +55,11 @@ class Mame < Formula
   fails_with gcc: "5"
   fails_with gcc: "6"
 
+  patch do
```
Comment (SMillerDev) [inline Formula/mame.rb:58]: Why is this needed and when can it be removed?

Comment (danielnachun) [reply Formula/mame.rb:60]: I added a comment explaining that this should be removed with the next release.

**Review by danielnachun — APPROVED** [review]
